fix(card): add `color` property to `md-card.md-THEME_NAME-theme` style

ilovett opened the pull request:
Depends on #7693
Depends on #7686

Closes #7687

ilovett commented:
preview:

![image](https://cloud.githubusercontent.com/assets/1776100/13937518/af2df358-ef82-11e5-9b91-2add003b3e69.png)

DevVersion commented on src/components/card/demoBasicUsage/index.html:
```diff
@@ -1,7 +1,7 @@
   <md-content class="md-padding" layout-xs="column" layout="row">
     <div flex-xs flex-gt-xs="50" layout="column">
-      <md-card>
+      <md-card md-theme="dark">
```
I'm not sure, it looks a bit weird in the demos, when the first card is dark. In my opinion it would be awesome if we have a checkbox for toggle the dark theme.

ilovett replied:
Yes I like that and perhaps with multiple themes with different light/dark contrasting values

DevVersion replied:
Yes, I suggest only `default` and `default-dark`

DevVersion commented:
@ilovett Tested it locally and it doesn't work, because all card demos use the same module, and that's a bug. (See #7693).

ilovett commented:
Cool thanks, should I write "Depends on" or anything to signify that?

DevVersion commented:
@ilovett No there is no need to do that :)

ilovett commented:
Here's new preview

![image](https://cloud.githubusercontent.com/assets/1776100/13972056/98418058-f052-11e5-8265-41998072ffff.png)

ilovett commented:
Want to mention that I'm using grey backgroundPalette due to #7600.

Once #7686 is merged I will happily add different color dark themes.

DevVersion commented:
@ilovett Great, so this PR is depending on two others? Maybe you can update the description if so.

ilovett commented:
Rebased on upstream/master after merge of #7686 , added different background colors.

![image](https://cloud.githubusercontent.com/assets/1776100/14193281/4be92276-f758-11e5-9d44-ab284c2781ab.png)

ilovett commented:
While looking into this I noticed perhaps the color and background color SHOULD be something more like
```
color: {{ background-contrast }}
background: {{ background-color }}
```
Where the contrast ends up being the calculated transparent white or #FFF -- and then in addition, if palettes have `contrastStrongDarkColors` or `contrastStrongLightColors`

I would suggest a dynamic weight variable:

```
font-weight: {{ contrast-weight }};
```

Where weight is either `500` or `800` or `bold`, etc.

I read somewhere a while back that the entirety of dark themes will be revisited soon, so I've avoided making bigger changes than just changing the foreground text color.
